WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/addons/AddonRepositoryTest.kt

```diff
@@ -0,0 +1,132 @@
+package com.woocommerce.android.ui.products.addons
+
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import com.woocommerce.android.extensions.unwrap
+import com.woocommerce.android.tools.SelectedSite
+import com.woocommerce.android.ui.products.addons.AddonTestFixtures.defaultOrderAttributes
+import com.woocommerce.android.ui.products.addons.AddonTestFixtures.defaultWCProductModel
+import com.woocommerce.android.ui.products.addons.AddonTestFixtures.defaultWCOrderItemList
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.WCOrderModel
+import org.wordpress.android.fluxc.model.order.OrderIdentifier
+import org.wordpress.android.fluxc.store.WCOrderStore
+import org.wordpress.android.fluxc.store.WCProductStore
+import kotlin.test.fail
+
+class AddonRepositoryTest {
+    private lateinit var repositoryUnderTest: AddonRepository
+
+    private lateinit var orderStoreMock: WCOrderStore
+    private lateinit var productStoreMock: WCProductStore
+    private lateinit var selectedSiteMock: SelectedSite
+    private lateinit var siteModelMock: SiteModel
+
+    private val localSiteID = 321
+    private val remoteOrderID = 123L
+    private val remoteProductID = 333L
+
+    @Before
+    fun setUp() {
+        siteModelMock = mock {
+            on { id }.doReturn(321)
+        }
+        orderStoreMock = mock()
+        productStoreMock = mock()
+        selectedSiteMock = mock {
+            on { get() }.doReturn(siteModelMock)
+        }
+
+        repositoryUnderTest = AddonRepository(
+            orderStoreMock,
+            productStoreMock,
+            selectedSiteMock
+        )
+    }
+
+    @Test
+    fun `fetchOrderAddonsData should return both order addons and product addons data`() {
+        configureSuccessfulOrderResponse()
+        configureSuccessfulProductResponse()
+
+        repositoryUnderTest.fetchOrderAddonsData(123, 999, 333)
+            ?.unwrap { productAddons, orderAddons ->
```

Comment:
⛏️  Does it make sense to use Kotlin's destructuring declarations here? So basically you could refactor this to 
```suggestion
            ?.let { (productAddons, orderAddons) ->
```

this way we don't have to introduce `unwrap` method. WDYT? Will it work for other cases too?

Reply:
Great call! Makes perfect sense, changed here: https://github.com/woocommerce/woocommerce-android/pull/4590/commits/11967bc36c827de950e5d4fcd5b6c7f6f4d462a5